### PR TITLE
Squads Overhaul Part 1

### PR DESCRIPTION
**Pull request by realforest2001**
## About The Pull Request
Updates the way Squads are handled, providing ERT squads and checks for what squad can be seen by whom.

## Why It's Good For The Game
Code is currently poorly optimized and linear, updating it would allow for more flexibility and allows for extra utility with ERTs.

## Changelog

:cl:
qol: Renamed marine cryo equipment presets and ERTs to the same standards and reduce confusion.
expansion: Squads now have a number of config variables to determine if they can be transferred into, overwatched, or used by your faction.
add: Echo Squad is now activated from the Groundside Operations console, rather than being active from the start of the round.
add: Squad Leaders now have a HUD icon even when not active squad leader, being assigned as (acting) Squad Leader uses same icon as present.
add: Cryorines are now automatically added to a Reserves squad.
add: MARSOC are now automatically added to a unique squad.
add: Added MARSOC Team Leader and MARSOC Officer roles.
add: Added the option to choose between covert MARSOC and standard ops. This changes if they use codenames or not.
add: MARSOC now have HUD Icons.
add: Squads now have defined types; Squad, Team and Cell. These determine what squad leaders are called in overwatch and the icons used for aSLs in that squad.
code: Working on 1-2 squads for PMCs and 4 for UPP.
/:cl:

### CODE REVIEW

**Comment by Watermelon914** [comment]
avoid adding too many tags to your changelog.

Changelog only needs to contain player-facing information

**Comment by Watermelon914** [comment]
I also think you should fill out the necessary sections, especially 'Why It's Good For The Game'

**Review by Agameofscones — APPROVED** [review]
icon changes are OK

File: code/controllers/subsystem/communications.dm [hunk 1]
```diff
@@ -158,25 +126,28 @@ var/list/radiochannels = list(
...
 // central command channels, i.e deathsquid & response teams
-#define CENT_FREQS list(ERT_FREQ, DTH_FREQ, PMC_FREQ, DUT_FREQ, YAUT_FREQ, HC_FREQ)
+#define CENT_FREQS list(ERT_FREQ, DTH_FREQ, PMC_FREQ, DUT_FREQ, YAUT_FREQ)
 
 // Antag channels, i.e. Syndicate
 #define ANTAG_FREQS list()
 
 //Depts - used for colors in headset.dm, as well as deciding what the marine comms tower can listen into
-#define DEPT_FREQS list(COMM_FREQ, MED_FREQ, ENG_FREQ, SEC_FREQ, ERT_FREQ, DTH_FREQ, ALPHA_FREQ, BRAVO_FREQ, CHARLIE_FREQ, DELTA_FREQ, ECHO_FREQ, SUP_FREQ, JTAC_FREQ, TACTICS_FREQ, WY_FREQ)
+#define DEPT_FREQS list(COMM_FREQ, MED_FREQ, ENG_FREQ, SEC_FREQ, ERT_FREQ, DTH_FREQ, COLONY_FREQ, ALPHA_FREQ, BRAVO_FREQ, CHARLIE_FREQ, DELTA_FREQ, ECHO_FREQ, CRYO_FREQ, SUP_FREQ, JTAC_FREQ, TACTICS_FREQ, WY_FREQ, HC_FREQ, MARSOC_FREQ)
```
Comment (Geevies) [inline code/controllers/subsystem/communications.dm:150]: colony freq shoudn't be on here, because this is the marine radio tower, same with HC and marsoc, and also ERT and DTH now that i see that. WY can stay cuz of CL

File: code/game/jobs/job/marine/squads.dm [hunk 2]
```diff
@@ -55,43 +83,136 @@
...
+/datum/squad/upp/four
+	name = "UPPS4"
+	color = 4
+
+/datum/squad/upp/kdo
+	name = "UPPKdo"
+	color = 6
+	squad_type = "Team"
+	locked = TRUE
+//###############################
```
Comment (Geevies) [inline code/game/jobs/job/marine/squads.dm:176]: the whitespaces between the lines here are annoying, just put a single whitespace between every entry instead of sometimes 1, sometimes 0

File: code/game/jobs/job/marine/squads.dm [hunk 3]
```diff
@@ -55,43 +83,136 @@
...
+/datum/squad/clf/one
+	name = "Python"
+/datum/squad/clf/two
+	name = "Viper"
+/datum/squad/clf/three
+	name = "Cobra"
+/datum/squad/clf/four
+	name = "Boa"
+//###############################
+/datum/squad/freelance
```
Comment (Geevies) [inline code/game/jobs/job/marine/squads.dm:212]: dont implement these if they aren't being used yet, write them down on a notepad or something

File: code/game/machinery/computer/groundside_operations.dm [hunk 4]
```diff
@@ -23,6 +23,8 @@
 	var/tacmap_base_type = TACMAP_BASE_OCCLUDED
 	var/tacmap_additional_parameter = null
 	var/minimap_name = "Marine Minimap"
+	var/echo = FALSE//If echo squad is active or not
```
Comment (Geevies) [inline code/game/machinery/computer/groundside_operations.dm:26]: this should be a global var

File: code/game/objects/items/devices/radio/headset.dm [hunk 5]
```diff
@@ -609,13 +614,57 @@
...
+/obj/item/device/radio/headset/almayer/marine/cryo/med
+	name = "marine reserves corpsman radio headset"
+	desc = "This is used by the marine Reserve combat medics. To access the medical channel, use :m. When worn, grants access to Squad Leader tracker. Click tracker with empty hand to open Squad Info window."
+	initial_keys = list(/obj/item/device/encryptionkey/public, /obj/item/device/encryptionkey/med)
+
+
+
+
+
+
```
Comment (Geevies) [inline code/game/objects/items/devices/radio/headset.dm:651]: ton of unnecessary whitespace, use two lines, three at most

File: code/modules/clothing/suits/marine_armor.dm [hunk 6]
```diff
@@ -24,19 +24,21 @@
...
 var/list/glovemarkings = list()
-var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200), rgb(65,72,200), rgb(103,214,146))
-var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150,255), rgb(130,140,255), rgb(103,214,146))
+var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200), rgb(65,72,200), rgb(103,214,146), rgb(133, 115, 75), rgb(64, 0, 0))
+var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150,255), rgb(130,140,255), rgb(103,214,146), rgb(133, 115, 75), rgb(64, 0, 0))
 
 /proc/initialize_marine_armor()
 	var/i
-	for(i=1, i<6, i++)
+	for(i=1, i<8, i++)
```
Comment (Geevies) [inline code/modules/clothing/suits/marine_armor.dm:41]: try replacing the 8 with length(squad_colors)

File: code/modules/cm_marines/overwatch.dm [hunk 7]
```diff
@@ -788,7 +794,12 @@
 		to_chat(usr, "[icon2html(src, usr)] [SPAN_WARNING("Transfer aborted. [transfer_marine] isn't wearing an ID.")]")
 		return
 
-	var/new_squad_name = tgui_input_list(usr, "Choose the marine's new squad", "Squad Selection", ROLES_SQUAD_ALL)
+	var/list/available_squads = list()
+	for(var/datum/squad/Squad in ROLES_SQUAD_ALL)
+		if(!Squad.locked && Squad.faction == faction)
+			available_squads += Squad
```
Comment (Geevies) [inline code/modules/cm_marines/overwatch.dm:800]:
```suggestion
	for(var/datum/squad/squad as anything in ROLES_SQUAD_ALL)
		if(!squad.locked && squad.faction == faction)
			available_squads += squad
```

**Review by Geevies — CHANGES_REQUESTED** [review]

File: code/modules/cm_marines/marines_consoles.dm [hunk 8]
```diff
@@ -583,7 +584,7 @@
 	var/list/data = list()
 	var/list/squads = list()
 	for(var/datum/squad/S in RoleAuthority.squads)
-		if(S.usable)
+		if(!(S.name == "Root") && !S.locked && S.active && (S.faction == faction))
```
Comment (Geevies) [inline code/modules/cm_marines/marines_consoles.dm:587]:
```suggestion
		if(S.name != "Root" && !S.locked && S.active && S.faction == faction)
```

File: code/modules/cm_marines/overwatch.dm [hunk 9]
```diff
@@ -449,7 +455,7 @@
 				else
 					var/list/squad_list = list()
 					for(var/datum/squad/S in RoleAuthority.squads)
-						if(S.usable && !S.overwatch_officer)
+						if((S.active && S.faction == faction) && !S.overwatch_officer && (S.name != "Root"))
```
Comment (Geevies) [inline code/modules/cm_marines/overwatch.dm:458]:
```suggestion
						if(S.active && !S.overwatch_officer && S.faction == faction && S.name != "Root")
```

File: code/modules/cm_marines/overwatch.dm [hunk 10]
```diff
@@ -788,10 +794,14 @@
 		to_chat(usr, "[icon2html(src, usr)] [SPAN_WARNING("Transfer aborted. [transfer_marine] isn't wearing an ID.")]")
 		return
 
-	var/new_squad_name = tgui_input_list(usr, "Choose the marine's new squad", "Squad Selection", ROLES_SQUAD_ALL)
-	if(!new_squad_name || S != current_squad)
+	var/list/available_squads = list()
+	for(var/datum/squad/squad as anything in RoleAuthority.squads)
+		if(squad.active && !squad.locked && (squad.faction == faction) && (squad.name != "Root"))
```
Comment (Geevies) [inline code/modules/cm_marines/overwatch.dm:799]:
```suggestion
		if(squad.active && !squad.locked && squad.faction == faction && squad.name != "Root")
```

File: code/modules/cm_marines/overwatch.dm [hunk 11]
```diff
@@ -788,10 +794,14 @@
...
+			available_squads += squad
+
+	var/datum/squad/new_squad = tgui_input_list(usr, "Choose the marine's new squad", "Squad Selection", available_squads)
+	if(!new_squad || S != current_squad)
```
Comment (Geevies) [inline code/modules/cm_marines/overwatch.dm:803]: what's that current squad check for?

**Review by Geevies — CHANGES_REQUESTED** [review]

**Comment by Geevies** [comment]
also check that marsoc/hc headsets have ignore_z set

File: code/modules/cm_marines/overwatch.dm (hunk 11 above)
Comment (realforest2001) [reply code/modules/cm_marines/overwatch.dm:803]: This isn't a me one, pre-existing. It's to check if they've already been moved, mid move. Prevents it crashing by somehow happening twice.

File: ColonialMarinesALPHA.dme [hunk 12]
```diff
@@ -377,7 +377,6 @@
 #include "code\datums\emergency_calls\clf.dm"
 #include "code\datums\emergency_calls\colonist.dm"
 #include "code\datums\emergency_calls\cryo_marines.dm"
-#include "code\datums\emergency_calls\cryo_marines_heavy.dm"
```
Comment (realforest2001) [inline ColonialMarinesALPHA.dme:380]: `#include "code\datums\emergency_calls\cryo_marines_heavy.dm"`
